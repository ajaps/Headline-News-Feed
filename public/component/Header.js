import React from 'react';
import PropTypes from 'prop-types';

import * as myActions from '../actions/HeadlineActions';


/**
 * Represents language preference and logout button
 */
export default class Header extends React.Component {

/**
* sends an action to change language preference
* @param {String} lang The preffered language.
* @returns {void}
*/
  setLanguage(lang) {
    myActions.setLanguage(lang);
  }

/**
 * @returns {component} A component with Header details like language and sign out functionality.
 */
  render() {
    return (
      <nav className="navbar navbar-inverse navbar-fixed-top">
        <div className="container-fluid">
          <div className="navbar-header">
            <button type="button" className="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
              <span className="sr-only">Toggle navigation</span>
              <span className="icon-bar" />
              <span className="icon-bar" />
              <span className="icon-bar" />
            </button>
            <a className="navbar-brand" href="/">Headlines RSS Feed</a>
          </div>
          <div id="navbar" className="navbar-collapse collapse">
            <ul className="nav navbar-nav navbar-right">
              <li><a cursor="pointer" onClick={this.setLanguage.bind(this, 'en')}>English</a></li>
              <li><a className="link_cursor" onClick={this.setLanguage.bind(this, 'de')}>German</a></li>
              <li><a className="link_cursor" onClick={this.setLanguage.bind(this, 'fr')}>French</a></li>
              <li><a onClick={this.props.logout} id="logoutBtn">Logout</a></li>
            </ul>
          </div>
        </div>
      </nav>
    );
  }
}

Header.propTypes = {
  logout: PropTypes.func.isRequired,
};
