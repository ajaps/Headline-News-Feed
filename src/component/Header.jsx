import React from 'react';
import PropTypes from 'prop-types';

/**
 * Represents the Header section, contains language preference and logout button
 */
export default class Header extends React.Component {

/**
* sends language selection to props
* @param {String} lang The preffered language.
* @returns {void}
*/
  setLanguage(lang) {
    this.props.setLan(lang);
  }

  /**
  * @returns {component} A component with Header details like language and sign out functionality.
  */
  render() {
    const allLanguage = this.props.allLanguage;
    // iterate through Language object
    const languageComponent = allLanguage.map((languageItem) => {
      const key = languageItem.key;
      return <li key={key}><a onClick={this.setLanguage.bind(this, languageItem.key)}>
      {languageItem.text}</a></li>;
    });

    return (
      <nav className="navbar navbar-inverse navbar-fixed-top">
        <div className="container-fluid">
          <div className="navbar-header">
            <a className="navbar-brand" href="/">Headlines RSS Feed</a>
          </div>
          <div id="navbar" className="navbar-collapse collapse">
            <ul className="nav navbar-nav navbar-right">
              {languageComponent}
              <li><a onClick={this.props.logout} id="logoutBtn">Logout</a></li>
            </ul>
          </div>
        </div>
      </nav>
    );
  }
}

Header.propTypes = {
  logout: PropTypes.func,
};
