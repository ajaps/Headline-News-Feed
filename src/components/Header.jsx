import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { logout } from '../config/authentication';

/**
 * Represents the Header section, contains language preference and logout button
 */
export default class Header extends React.Component {
  constructor() {
    super();
    this.initiateLogout = this.initiateLogout.bind(this);
  }

/**
 * Logs out from application by calling logout function in config file
 * @return {void}
 */
  initiateLogout() {
    logout();
  }

  render() {
    const showLogoutBtn = classNames({
      hidden: !(this.props.containLogoutBtn)
    });

    return (
      <nav className="navbar navbar-inverse navbar-fixed-top">
        <div className="container-fluid">
          <div className="navbar-header">
            <a className="navbar-brand" href="/">Headlines RSS Feed</a>
          </div>
          <div id="navbar" className="navbar-collapse collapse">
            <ol className="nav navbar-nav navbar-right">
              <li className={showLogoutBtn}><a className="disableClick">
                {/* {`Welcome:   ${ }`} */}
              </a></li>
              <li id="logoutBtn" className={showLogoutBtn}>
                <a onClick={this.initiateLogout}>Logout</a>
              </li>
            </ol>
          </div>
        </div>
      </nav>
    );
  }
}

Header.propTypes = {
  containLogoutBtn: PropTypes.bool,
  userName: PropTypes.string
};

Header.defaultProps = {
  containLogoutBtn: false,
  userName: ''
};
