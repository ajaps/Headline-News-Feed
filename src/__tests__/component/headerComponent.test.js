import { mount, shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Header from '../../component/Header';

describe('Header', () => {
  let app;
  beforeEach(() => {
    const allLanguages = [{ key: 'ig', text: 'Igbo' }, { key: 'yr', text: 'Yoruba' }];
    app = shallow(<Header allLanguage={allLanguages} setLan={() => 'y' } />);
  });

  it('should render as expected', () => {
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const allLanguages = [{ key: 'ig', text: 'Igbo' }, { key: 'yr', text: 'Yoruba' }];
    shallow(<Header key={allLanguages.key} allLanguage={ allLanguages } />);
  });

  it('is a function', () => {
    app.instance().setLanguage();
  });

  it('contains Igbo language clickable element', () => {
    expect(app.exists(<li>Igbo</li>)).toBe(true);
  });

  it('contains Yoruba language clickable element', () => {
    expect(app.exists(<li>Yoruba</li>)).toBe(true);
  });

  it('contains Logout clickable element', () => {
    expect(app.exists(<li>Logout</li>)).toBe(true);
  });

  it('recieves Two(2) props', () => {
    expect(Object.keys(app.props()).length).toBe(2);
  });

  it('to contain 3 li(list item) when two object are passed', () => {
    expect(app.find('li').length).toEqual(3);
  });
});
