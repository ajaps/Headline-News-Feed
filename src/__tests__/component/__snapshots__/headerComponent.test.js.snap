// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render as expected 1`] = `
<nav
  className="navbar navbar-inverse navbar-fixed-top"
>
  <div
    className="container-fluid"
  >
    <div
      className="navbar-header"
    >
      <button
        aria-controls="navbar"
        aria-expanded="false"
        className="navbar-toggle collapsed"
        data-target="#navbar"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="sr-only"
        >
          Toggle navigation
        </span>
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </button>
      <a
        className="navbar-brand"
        href="/"
      >
        Headlines RSS Feed
      </a>
    </div>
    <div
      className="navbar-collapse collapse"
      id="navbar"
    >
      <ul
        className="nav navbar-nav navbar-right"
      >
        <li>
          <a
            onClick={[Function]}
          >
            Igbo
          </a>
        </li>
        <li>
          <a
            onClick={[Function]}
          >
            Yoruba
          </a>
        </li>
        <li>
          <a
            id="logoutBtn"
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
